app/routes/session.py

```diff
@@ -169,9 +181,26 @@ def _set_questionnaire_supplementary_data(
         survey_id=metadata["survey_id"],
         sds_dataset_id=new_sds_dataset_id,
     )
+    _validate_supplementary_data_lists(
+        supplementary_data=supplementary_data["data"], schema=schema
+    )
     questionnaire_store.set_supplementary_data(supplementary_data["data"])
 
 
+def _validate_supplementary_data_lists(
+    *, supplementary_data: dict, schema: QuestionnaireSchema
+) -> None:
+    """
+    Validates that any lists the schema requires (which are those in the supplementary_data.lists property)
+    are included in the supplementary data
+    """
+    supplementary_lists = set(supplementary_data.get("items", {}).keys())
+    if missing := schema.supplementary_lists - supplementary_lists:
+        raise ValidationError(
+            f"Supplementary data does not include the following lists required for the schema: {', '.join(missing)}"
+        )
```

Comment:
If it is keys in schema that aren't in SDS, then this needs to be flipped?.

Reply:
I think I've got this the right way round? missing is { any supplementary lists the schema requires that are not in the actual supplementary data lists }

Reply:
If it looks wrong from the naming happy to modify

Reply:
Nope, all good. I misread 👍